perf/lua/1mops_write.lua

```diff
@@ -0,0 +1,294 @@
+            require('fiber').sleep(0.1)
+        end
+    else
+        print('Incorrect number of nodes \"' .. arg[1] .. '\" must be 1..31')
+        os.exit(1)
+    end
+end
+
+local space
+local done = false
+local err
+
+if (test_qsync) then
+    box.cfg{replication_synchro_quorum = nodes}
+    print('promoting')
+    box.ctl.promote()
+    print('done')
+    space, err = box.schema.create_space('test',
+                                         {engine = engine, is_sync = true})
+else
+    space, err = box.schema.create_space('test',
+                                         {engine = engine})
+end
+if space == false then
+    print('error creating space', json.encode(err))
+    os.exit(1)
+end
+local res, err = pcall(space.create_index, space, 'pk', index_config)
+if (res ~= true) then
+    print('error creating index: ', json.encode(err))
+    os.exit(1)
+end
+
+-- THE load fiber
+local function fiber_load(start, s)
+    start = start % 1000000 -- limit the size of space to 1M elements
+    for _ = 1, trans_per_fiber do
+        box.begin()
+        for _ = 1, ops_per_txn do
+            s:replace{start}
+            start = start + 1
+        end
+        box.commit()
+    end
+end
+
+-- fiber storage to join them
+local fibers_storage = {}
+
+collectgarbage()
+collectgarbage()
+
+-- start timer
+local timer_begin = {
+    clock.time(),
+    clock.proc()
+}
+
+local max_rps = 0
+local ops_done = 0
+
+-- start a fiber to perform the peak RPS measurement
+-- skip the warmup if any and restart the timer
+fiber.new(function()
+    fiber.new(function()
+        while true do
+            local prev_t = clock.time()
+            local prev_lsn = box.info.lsn
+            fiber.sleep(0.1)
+            local rps = (box.info.lsn - prev_lsn) / (clock.time() - prev_t)
+            if (rps > max_rps) then
+                max_rps=rps
+            end
+        end
+    end)
+    if warmup_thr > 0 then
+        io.write('Warmup... ')
+        io.flush()
+        while box.info.lsn < num_ops / 100 * warmup_thr do
+            fiber.sleep(0.01)
+        end
+        ops_done = box.info.lsn
+        print('done, lsn: ', ops_done)
+        timer_begin = {
+            clock.time(),
+            clock.proc()
+        }
+    end
+end)
+
+-- start fibers for the main load
+for i = 1, num_fibers do
+    fibers_storage[i] = fiber.create(fiber_load, i*num_ops, space)
+    if (fibers_storage[i]:status() ~= 'dead') then
+        fibers_storage[i]:wakeup() -- needed for backward compatibility with 1.7
+    end
+end
+
+-- wait for all fibers to finish
+for i = 1, num_fibers do
+    while fibers_storage[i]:status() ~= 'dead' do
+        fiber.yield()
+    end -- the loop is needed for backward compatibility with 1.7
+end
+
+-- stop timer for master
+local len = {
+    clock.time()-timer_begin[1],
+    clock.proc()-timer_begin[2]
+}
+
+ops_done = box.info.lsn - ops_done
+
+print(string.format('master done %d ops in time: %f, cpu: %f',
```

Comment:
It would be perfect to have an option for TSV/CSV output to process the result via Perl script or load it into Excel to calculate some statistics.

Reply:
Followed the rules of [compare.lua](https://github.com/tarantool/tarantool/blob/master/perf/lua/compare.lua) - see the comment below